domain/storageprovisioning/state/state.go

```diff
@@ -591,6 +592,59 @@ WHERE  storage_id = $storageID.storage_id`, input, dbVal)
 	return dbVal.UUID, nil
 }
 
+// GetStorageAttachmentInfo returns information about a storage attachment for
+// the given storage attachment UUID.
+//
+// The following errors may be returned:
+// - [storageprovisioningerrors.StorageAttachmentNotFound] when the storage
+// attachment does not exist.
+func (st *State) GetStorageAttachmentInfo(
+	ctx context.Context, uuid string,
+) (storageprovisioning.StorageAttachmentInfo, error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return storageprovisioning.StorageAttachmentInfo{}, errors.Capture(err)
+	}
+
+	input := storageAttachmentUUID{UUID: uuid}
+	var dbVal storageAttachmentInfo
+	stmt, err := st.Prepare(`
+SELECT    (si.life_id, cs.storage_kind_id) AS (&storageAttachmentInfo.*),
```

Comment:
Need to be very careful here. Charm storage kind is not the same as storage kind. We actually missed this in our modelling originally and I have proposed adding it in #20550. I would default this value off for the moment and put a TODO if this PR lands before mine.